### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by Vianpyro**
Here's a comprehensive Pull Request description based on your changes:

---

## Brief Description of Changes  
This PR introduces several enhancements to the testing infrastructure and utility functions, including:  
- Automated test execution via VS Code's RunOnSave extension  
- GitHub Actions CI workflow for cross-platform testing  
- Email validation utility function  
- Improved email masking function  
- Comprehensive test coverage for authentication routes and utility functions  

## Detailed Changes  

1. **Development Workflow Improvements**  
   - Added `emeraldwalk.RunOnSave` configuration to automatically run pytest on test file saves  
   ```json
   "emeraldwalk.runonsave": {
       "commands": [
           {
               "match": "tests[/\\\\](.*[/\\\\])?test_.*\\.py$",
               "cmd": "python3 -m pytest '${relativeFile}' -v",
               "autoShowOutputPanel": "error"
           }
       ]
   }
   ```

2. **CI/CD Pipeline**  
   - Implemented GitHub Actions workflow for automated testing across multiple platforms (Linux, macOS, Windows)  
   - Includes test coverage reporting with pytest-cov  

3. **Utility Functions**  
   - Added `validate_email()` function with regex pattern for email validation  
   ```python
   def validate_email(email: str) -> bool:
       pattern = r"^[a-zA-Z0-9._+-]+@[a-zA-Z0-9-]+(\.[a-zA-Z0-9-]+)*\.[a-zA-Z]{2,}$"
       return bool(re.match(pattern, email))
   ```
   - Refactored `mask_email()` function for better handling of complex email formats  
     - Simplified logic using string splitting  
     - Improved masking of local parts and domain components  

4. **Test Coverage**  
   - Added comprehensive test suites for:  
     - Authentication routes (login, registration, token refresh)  
     - Utility functions (email encryption/decryption, validation, masking, password hashing)  

## Objective of Changes  

- **Fixes email masking issues** for non-standard email formats  
- **Resolves the problem of** manual test execution during development  
- **Enhances code quality** through automated testing on multiple platforms  
- **Improves security** with robust email validation for the registration endpoint  

## Verification Steps  

1. **Local Testing:**  
   - Save any test file in the `tests/` directory - pytest should automatically run  
   - Manually verify email masking:  
     ```python
     mask_email("test.user+123@sub.example.com")  # Should return "t**t.u**r+1*3@s*b.e*****e.c*m"
     ```

2. **CI Verification:**  
   - Check GitHub Actions runs complete successfully on all platforms  
   - Verify coverage reports are generated  

3. **Functionality Tests:**  
   - Test registration endpoint with various email formats:  
     - Valid: `user@example.com`  
     - Invalid: `user@.com`, `user@example`  

## Notes for Reviewer  

- Focus on the email masking logic changes - the new version handles more edge cases  
- Verify the test coverage for new utility functions  
- Check CI workflow configuration for platform-specific test commands

### CODE REVIEW

File: tests/test_routes/test_authentication/test_register.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    response = client.post("/register", json=data)
+
+    assert response.status_code == 400
+    assert response.json["message"] == "Username, email, and password are required"
+
+
+def test_missing_username(client: FlaskClient, sample_username, sample_email):
+    """Test registration with missing username"""
+    data = {
+        "username": sample_username,  # Missing username
```
[the remark this answers is not in the document]
Comment (Vianpyro) [reply tests/test_routes/test_authentication/test_register.py]: This was fixed manually to include modifications to the function arguments
